### DISCUSSION TITLE
Is there any way to avoid Cartesian Product problem with EntityViews?

### DISCUSSION THREAD

**Discussion by sanemain** [Blazebit/blaze-persistence#1246] [Q&A] [answered]
Hi, as far as I understand from the EntityView documentation, the DTO creation always uses JOINs underneath. I can see this avoids the n+1 select problem, but are not we now exposed to Cartesian Product problem? Is there a way to use separate selects for some parts while constructing the DTOs?

Thanks.

**Comment by beikov** [comment, marked as answer]
Hi,

yeah you can switch the fetch strategy as outlined in the documentation: https://persistence.blazebit.com/documentation/entity-view/manual/en_US/index.html#anchor-fetch-strategies

Usually, the `MULTISET` fetch strategy is the one that performs best, but there might be reasons why it can't be used.

**Reply by sanemain** [reply]
Thanks for the blazing fast response :)
Obviously, I missed that part. You might also want to include some notes about fetch strategies in quick start tutorials or feature notes. This clears some of my doubts about the project, now I will take a more serious look at the documentation for sure. Thanks for all the efforts for putting up this library.

**Reply by beikov** [reply]
No worries, enjoy it :)
